compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/PropertyReferenceLowering.kt

```diff
@@ -165,20 +165,17 @@ internal class PropertyReferenceLowering(val context: JvmBackendContext) : Class
                 cachedKProperty(expression)
 
             private fun cachedKProperty(expression: IrCallableReference): IrExpression {
-                // Reflected implementation does not support partial application; also, cannot cache instances with arguments.
-                if (expression.dispatchReceiver != null || expression.extensionReceiver != null)
-                    return createSpecializedKProperty(expression)
-
                 // For delegated properties, the getter and setter contain a reference each as the second argument to getValue
                 // and setValue. Since it's highly unlikely that anyone will call get/set on these, optimize for space.
                 val useReflectedImpl = expression.origin == IrStatementOrigin.PROPERTY_REFERENCE_FOR_DELEGATE
+
+                // Reflected implementation does not support partial application; also, cannot cache instances with arguments.
+                if (!useReflectedImpl || expression.dispatchReceiver != null || expression.extensionReceiver != null)
```

Comment:
I think property reference for delegate is guaranteed not to have receivers, so it should be enough to check `!useReflectedImpl` (and inline it)

Reply:
Good to know! Fixed.